tests/testsuite/jobserver.rs

```diff
@@ -212,33 +215,106 @@ test-runner:
         .env("CARGO", cargo_exe())
         .arg("test-runner")
         .arg("-j2")
-        .with_stderr_contains("[..]this is a runner[..]")
+        .with_stderr_data(str![[r#"
+[WARNING] no edition set: defaulting to the 2015 edition while the latest is 2021
+[FINISHED] `test` profile [unoptimized + debuginfo] target(s) in [ELAPSED]s
+[RUNNING] unittests src/lib.rs (target/debug/deps/cargo_jobserver_check-[HASH][EXE])
+this is a runner
+
+"#]])
         .run();
 
     // but not from `-j` flag
     p.cargo("run -j2")
         .with_status(101)
-        .with_stderr_contains("[..]no jobserver from env[..]")
+        .with_stderr_data(str![[r#"
+[WARNING] no edition set: defaulting to the 2015 edition while the latest is 2021
```

Comment:
In a commit before this one, please set the edition in the package, likely to 2015.

Reply:
Thank you for adding `edition`.  I was hoping it would be in its own commit so the work is easier to track.  Would it be ok to go back and split it out?

Reply:
Thank you for this thoughtful reminder. Now it's a separated commit refer to #13505.